docs/tutorials/field_type/8_add_a_validation.md

```diff
@@ -149,38 +149,45 @@ private function isAuthorApproved($author, $validatorConfiguration)
 
 Earlier we validated the URL with a regular expression. Now, if the configuration of your Field Type's instance contains a TweetValueValidator key, you will check that the username in the status URL matches one of the valid authors.
 
-## Implement FormMapper
+## Implement FieldDefinitionFormMapperInterface in FormMapper
 
-We would like to offer a way to the user to input a list of authors (upon which the data will be validated). To achieve this, we will implement a FormMapper that allows us to define the necessary input field.
+We would like to offer a way to the user to input a list of authors (upon which the data will be validated). To achieve this, we will add additional functionality in FormMapper which allows us to define the necessary input field.
 This is a minimal example of our FormMapper:
 
-```php
+``` php
 // eZ/Publish/FieldType/Tweet/FormMapper.php
 
 namespace EzSystems\TweetFieldTypeBundle\eZ\Publish\FieldType\Tweet;
 
 use EzSystems\RepositoryForms\Data\FieldDefinitionData;
 use EzSystems\RepositoryForms\FieldType\FieldDefinitionFormMapperInterface;
+use EzSystems\RepositoryForms\FieldType\FieldValueFormMapperInterface;
 use Symfony\Component\Form\Extension\Core\Type\TextType;
 use Symfony\Component\Form\FormInterface;
 
-class FormMapper implements FieldDefinitionFormMapperInterface
+class FormMapper implements FieldDefinitionFormMapperInterface, FieldValueFormMapperInterface
 {
     public function mapFieldDefinitionForm(FormInterface $fieldDefinitionForm, FieldDefinitionData $data)
     {
         $fieldDefinitionForm
             ->add('authorList', TextType::class, [
                 'required' => false,
                 'property_path' => 'validatorConfiguration[TweetValueValidator][authorList]',
+                'translation_domain' => 'eztweet_fieldtype',
                 'label' => 'field_definition.eztweet.authorList'
             ]);
     }
+    
+    public function mapFieldValueForm(FormInterface $fieldForm, FieldData $data)
+    {
+        (...)
```

Comment:
Same comment: could we comment this parenthesis, because copy-paste is frequent in tutos?